src/fswAlgorithms/effectorInterfaces/prescribedTrans/prescribedTrans.c

```diff
@@ -0,0 +1,156 @@
+/*
+ ISC License
+
+ Copyright (c) 2023, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+*/
+
+/*! Import the module header file */
+#include "prescribedTrans.h"
+
+/*! Import other required files */
+#include <math.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "architecture/utilities/linearAlgebra.h"
+#include "architecture/utilities/macroDefinitions.h"
+
+/*! This method initializes the output message for this module. Other module parameters are also initialized.
+ @return void
+ @param configData The configuration data associated with this module
+ @param moduleID The module identifier
+ */
+void SelfInit_prescribedTrans(PrescribedTransConfig *configData, int64_t moduleID)
+{
+    // Initialize the module output message
+    PrescribedMotionMsg_C_init(&configData->prescribedMotionOutMsg);
+
+    // Set the initial time
```

Comment:
The module variables need to be initialized in Reset(), not SelfInit()

Reply:
Done.